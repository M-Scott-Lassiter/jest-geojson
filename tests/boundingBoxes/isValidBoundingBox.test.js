// This matcher works on all the same valid2DBoundingBox and valid3DBoundingBox cases.
// Accordingly, those specific tests check that both functions work.
// This test suite checks the array elements sizing because 2D and 3D have different acceptance criteria.
// Finally, it tests the unique snapshots.

describe('Invalid Use Cases', () => {
    describe('Expect to fail with incorrect number of array elements:', () => {
        test.each([
            [[]],
            [[20]],
            [[20, 30]],
            [[20, 30, 0]],
            [[20, 30, 0, 4, 0]],
            [[20, 30, 0, 20, 30, 0, 20]],
            [[20, 30, 0, 20, 30, 0, 20, 30, 0]]
        ])('expect(%p)', (badInput) => {
            expect(badInput).not.isValidBoundingBox()
        })
    })
})

describe('Error Snapshot Testing. Throws error:', () => {
    test('Valid use case passes', () => {
        expect(() =>
            expect([-20, 10, -10, 20]).not.isValidBoundingBox()
        ).toThrowErrorMatchingSnapshot()
    })

    test('Invalid input to matcher', () => {
        expect(() => expect(false).isValidBoundingBox()).toThrowErrorMatchingSnapshot()
    })

    test('Bounding box must be an array of either four or six elments', () => {
        expect(() => expect([0, 1, 2, 3, 4]).isValidBoundingBox()).toThrowErrorMatchingSnapshot()
    })
})
