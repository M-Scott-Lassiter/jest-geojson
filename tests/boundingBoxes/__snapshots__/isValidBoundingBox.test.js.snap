// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Snapshot Testing. Throws error: Bounding box must be an array of either four or six elments 1`] = `
"[2mexpect([22m[31m[west, south, (depth), east, north, (altitude)][39m[2m).isValidBoundingBox()[22m

Bounding box must be an array of either four or six elments.

Received:  [31m[0, 1, 2, 3, 4][39m"
`;

exports[`Error Snapshot Testing. Throws error: Invalid input to matcher 1`] = `
"[2mexpect([22m[31m[west, south, (depth), east, north, (altitude)][39m[2m).isValidBoundingBox()[22m

Bounding box must be an array.

Received:  [31mfalse[39m"
`;

exports[`Error Snapshot Testing. Throws error: Valid use case passes 1`] = `
"[2mexpect([22m[31m[west, south, (depth), east, north, (altitude)][39m[2m).not.isValidBoundingBox()[22m

Expected input to not be a four or six element bounding box array.

Received:  [31m[-20, 10, -10, 20][39m"
`;
